api/envoy/extensions/filters/network/meta_protocol_proxy/matcher/v3/matcher.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.meta_protocol_proxy.matcher.v3;
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.meta_protocol_proxy.matcher.v3";
+option java_outer_classname = "MatcherProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: Meta Protocol Proxy Route Matcher Configuration]
+
+// Used to match request service of the downstream request. Only relevant when the application
```

Comment:
How are these matchers used? It would be good to see some examples of config fragments somewhere.

Reply:
Get it.

Reply:
> How are these matchers used? It would be good to see some examples of config fragments somewhere.

@htuch examples can be found in the design doc: https://docs.google.com/document/d/1y_ALyjGp_H1_TRrrRxZxnqENISqeZI6eLn71FAFsZPk/edit#heading=h.pgxyh6f6oib4

Reply:
Would it be possible in this PR to have some config snippets? somewhere? I think that would really help.

Reply:
An example of the meta protocol route configuration with matcher API has been added to the route.proto.